### ISSUE TITLE
possible error in new_exposure function due to astropy version 

### ISSUE THREAD

**Issue by alxogm** [desihub/desisim#269]
Hi, I think the function new_exposure is not working correctly for astropy v.2.0. 

This caused an error in the  simulating-desi-spectra tutorial when I tried it my desktop, but it worked well in jupyter-dev.nersc. The same problem appear when running the integration_test
Problem was fixed by downgrade the astropy version to 1.3.3. Should  this be added to the wiki and/or tutorial?

**Comment by sbailey** [comment]
That is quite possible.  We don't yet support astropy 2.x.  Please go ahead and add that to the wiki installation instructions and any tutorials.  Please also post here what error you had so that when we do "upgrade" to 2.x, we'll make sure to check this examples in case the standard tests don't catch it.

**Comment by alxogm** [comment]
Ok, i'm adding it to wiki and tutorials. The problem I think is due to non defined units in astropy, the fix that was done for astropy 1.3 does not work for astropy 2. The error on simulating-desi-spectra tutorial  is as follow: 


```
fibermap, truth = new_exposure(flavor=flavor, nspec=nspec, seed=seed, night=night, 
                               expid=expid, tileid=None, exptime=None)

INFO:io.py:625:read_basis_templates: Reading /Users/alxogm/desi/spectro/templates/basis_templates/v2.3/lrg_templates_v1.3.fits
INFO:io.py:625:read_basis_templates: Reading /Users/alxogm/desi/spectro/templates/basis_templates/v2.3/star_templates_v2.1.fits
INFO:io.py:625:read_basis_templates: Reading /Users/alxogm/desi/spectro/templates/basis_templates/v2.3/elg_templates_v2.0.fits
INFO:obs.py:226:new_exposure: skyfile /Users/alxogm/desi/code/desimodel/data/spectra/spec-sky.dat
INFO:obs.py:297:new_exposure: Wrote /Users/alxogm/desi/spectro/sim/example/20170615/fibermap-00000000.fits
WARNING: AstropyDeprecationWarning: "clobber" was deprecated in version 2.0 and will be removed in a future version. Use argument "overwrite" instead. [desisim.io]
WARNING: The unit 'dex' could not be saved to FITS format [astropy.io.fits.convenience]
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-55382346f44f> in <module>()
      1 fibermap, truth = new_exposure(flavor=flavor, nspec=nspec, seed=seed, night=night, 
----> 2                                expid=expid, tileid=None, exptime=None)

~/miniconda3/envs/desi2/lib/python3.5/site-packages/desisim-0.19.0-py3.5.egg/desisim/obs.py in new_exposure(flavor, nspec, night, expid, tileid, airmass, exptime, seed, testslit, arc_lines_filename, flat_spectrum_filename)
    302     hdr['DATE-OBS'] = (time.strftime('%FT%T', dateobs), 'Start of exposure')
    303 
--> 304     simfile = io.write_simspec(meta, truth, expid, night, header=hdr)
    305     log.info('Wrote '+simfile)
    306 

~/miniconda3/envs/desi2/lib/python3.5/site-packages/desisim-0.19.0-py3.5.egg/desisim/io.py in write_simspec(meta, truth, expid, night, header, outfile)
    173 
    174         write_bintable(outfile, meta, header=None, extname="METADATA",
--> 175             comments=comments, units=units)
    176 
    177     return outfile

~/miniconda3/envs/desi2/lib/python3.5/site-packages/desispec-0.15.1-py3.5.egg/desispec/io/util.py in write_bintable(filename, data, header, comments, units, extname, clobber)
    120 
    121     # hdu = astropy.io.fits.BinTableHDU(outdata, header=header, name=extname)
--> 122     hdu = astropy.io.fits.convenience.table_to_hdu(outdata)
    123     if extname is not None:
    124         hdu.header['EXTNAME'] = extname

~/miniconda3/envs/desi2/lib/python3.5/site-packages/astropy/io/fits/convenience.py in table_to_hdu(table)
    511 
    512             # Try creating a Unit to issue a warning if the unit is not FITS compliant
--> 513             Unit(col.unit, format='fits', parse_strict='warn')
    514 
    515     for key, value in table.meta.items():

~/miniconda3/envs/desi2/lib/python3.5/site-packages/astropy/units/core.py in __call__(self, s, represents, format, namespace, doc, parse_strict)
   1824 
   1825         elif s is None:
-> 1826             raise TypeError("None is not a valid Unit")
   1827 
   1828         else:

TypeError: None is not a valid Unit
```



And when running the integration_tests is: 


```
CRITICAL:util.py:98:runcmd: FAILED newexp-desi --flavor dark --nspec 5 --night 20171030 --expid 2
Traceback (most recent call last):
  File "/Users/alxogm/miniconda3/envs/desi2/lib/python3.5/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Users/alxogm/miniconda3/envs/desi2/lib/python3.5/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/alxogm/miniconda3/envs/desi2/lib/python3.5/site-packages/desispec-0.15.1-py3.5.egg/desispec/test/old_integration_test.py", line 349, in <module>
    integration_test()
  File "/Users/alxogm/miniconda3/envs/desi2/lib/python3.5/site-packages/desispec-0.15.1-py3.5.egg/desispec/test/old_integration_test.py", line 98, in integration_test
    raise RuntimeError('pixsim newexp failed for {} exposure {}'.format(flavor, expid))
RuntimeError: pixsim newexp failed for dark exposure 2
```

**Comment by sbailey** [comment]
Works with astropy 2 now; closing.
